compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -673,33 +673,21 @@ trait Implicits { self: Typer =>
         EmptyTree
     }
 
+    /** Synthesize the tree for `'[T]` for an implicit `scala.quoted.Type[T]`.
+     *  `T` is deeply dealiassed to avoid references to local type aliases.
+     */
     def synthesizedTypeTag(formal: Type): Tree = {
       def quotedType(t: Type) = {
         if (StagingContext.level == 0)
           ctx.compilationUnit.needsStaging = true // We will need to run ReifyQuotes
         ref(defn.InternalQuoted_typeQuote).appliedToType(t)
       }
       formal.argInfos match {
-        case arg :: Nil if !arg.typeSymbol.is(Param) =>
-          object bindFreeVars extends TypeMap {
```

Comment:
This `TypeMap` is completely useless here. This is already performed by the PCP check in the `Staging` phase which emits errors with extra level information if the implicit is not found.